src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -610,9 +611,25 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
         {
             LogFilter logFilter = _blockchainBridge.GetFilter(filter.FromBlock, filter.ToBlock,
                 filter.Address, filter.Topics);
+
             IEnumerable<FilterLog> filterLogs = _blockchainBridge.GetLogs(logFilter, fromBlock, toBlock, cancellationToken);
 
-            return ResultWrapper<IEnumerable<FilterLog>>.Success(GetLogs(filterLogs, cancellationTokenSource));
+            ArrayPoolList<FilterLog> logs = new(_rpcConfig.MaxLogsPerResponse);
+
+            using (cancellationTokenSource)
+            {
+                foreach (FilterLog log in filterLogs)
+                {
+                    logs.Add(log);
+                    if (logs.Count > _rpcConfig.MaxLogsPerResponse)
```

Comment:
if `0` should we ignore this check?

Reply:
for the case if the `MaxLogsPerResponse` is `0` ?

Reply:
yes, should we treat 0 we may treat it as no limit